### PR TITLE
Implement authentication with devise

### PR DESCRIPTION
**Pull request by isaka-lumato**
## AUTHENTICATION WITH DEVISE
In this project, my partner and I used the devise gem for authenticating users in the app

- Users can register
- Login
- Logout
- Password is hashed and secure with the help of the devise gem

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (mmsesay) [inline app/models/user.rb:4]: - [ ] Normally, it is a good practice to verify your application users before they access some features of your application. The reason is that you don't want fake users. A remedy to this problem is to add `confirmable` to here and ensure that you send an email to the newly created user.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @isaka-lumato,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Removing unused files and folders. With over 1000 project files, it takes ages to clone. And I guess it also takes longer to push from your local machine to this repo.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (isaka-lumato) [reply app/models/user.rb:4]: thanks i just did this, but how do I delete those unused files? because if I delete the vendor folder the application breaks

Comment (mmsesay) [reply app/models/user.rb:4]: You can try to gitignore the vendor folder. 
Let me know if it works.

**Review by mmsesay — APPROVED** [review]
Hi @isaka-lumato,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**